Classes/Demo/Home/Models/HomePortfolioViewModel.swift

```diff
@@ -129,9 +138,16 @@ extension HomePortfolioViewModel {
         _ portfolioItem: TotalPortfolioItem,
         selectedPoint: ChartSelectedPointViewModel
     ) {
+        
+        var selectedPointSecondaryValue = selectedPoint.secondaryValue
+        
+        if let currency = try? portfolioItem.currency.secondaryValue?.unwrap(), currency.isAlgo {
+            selectedPointSecondaryValue = selectedPoint.primaryValue
+        }
+        
```

Comment:
You are using this code several times. Can you extract it as a method and reuse it?